Automattic/jetpack
# Super Cache: use "activate/install boost button" on the banner in other areas

The Boost Banner in WPSC has an "Activate Jetpack Boost" button that this PR modifies so it can be used in similar buttons in other places of the settings page such as an admin notice and a secondary "Try Boost" button.

## Proposed changes:

* Add the function wpsc_get_boost_migration_config() that fetches config data used by all migration buttons.
* Change JS so it triggers on the wpsc-install-boost-button class rather than a single ID.
* The button clicked (source) will be stored as data-source.
* That source is passed down to the activate function and on to PHP through an ajax request. It's passed on to Boost eventually.
* Use a wpsc-activate-boost-button class rather than an ID in a similar way to above.
* Modify the activateBoost function so it accepts a single "source" parameter. It also calls the wpsc_notify_migration_to_boost() function.
* 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:





Edit inc/boost.php and add this function to aid testing if #37797 hasn't been merged yet:
```
function wpsc_notify_migration_to_boost( $source ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
}
```
Load the WP Super Cache settings page and click on the "Activate Jetpack Boost" button.
You should be brought to the Boost settings page.
